src/components/views/settings/SetIdServer.js

```diff
@@ -127,28 +130,68 @@ export default class SetIdServer extends React.Component {
         return !!this.state.idServer && !this.state.busy;
     };
 
+    _continueTerms = (fullUrl) => {
+        MatrixClientPeg.get().setIdentityServerUrl(fullUrl);
+        localStorage.removeItem("mx_is_access_token");
+        localStorage.setItem("mx_is_url", fullUrl);
+        dis.dispatch({action: 'id_server_changed'});
+        this.setState({idServer: '', busy: false, error: null});
+    };
+
     _saveIdServer = async (e) => {
```

Comment:
...and this should change to something like:

```suggestion
    _checkIdServer = async (e) => {
```